GNUmakefile

```diff
@@ -366,11 +366,23 @@ proto-tools:
 codegen-tools:
 	@$(SHELL) $(CURDIR)/build-support/scripts/devtools.sh -codegen
 
+.PHONY: pre-commit-tools
+pre-commit-tools:
+	@$(SHELL) $(CURDIR)/build-support/scripts/devtools.sh -pre-commit
+
 .PHONY: deep-copy
 deep-copy: codegen-tools
 	@$(SHELL) $(CURDIR)/agent/structs/deep-copy.sh
 	@$(SHELL) $(CURDIR)/agent/proxycfg/deep-copy.sh
 
+# Devs shoudln't have to think about whether there is a hypen or not
+.PHONY: precommit
+precommit: pre-commit
+
```

Comment:
I like the fact that this is voluntary, but I guess once merged it need to be socialized so people install it locally.